Shared/Samples/Show viewshed from point on map/ShowViewshedFromPointOnMapView.swift

```diff
@@ -0,0 +1,202 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+/// Description
+struct ShowViewshedFromPointOnMapView: View {
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    /// The point on the screen where the user tapped.
+    @State private var tapScreenPoint: Point?
+    
+    /// The current geoprocessing status..
+    @State private var geoprocessingInProgress: Bool = false
+    
+    /// The data model for the sample.
+    @StateObject private var model = Model()
+    
+    var body: some View {
+        MapView(map: model.map, graphicsOverlays: [model.inputGraphicsOverlay, model.resultGraphicsOverlay])
+            .onSingleTapGesture { _, tapPoint in
+                self.tapScreenPoint = tapPoint
+            }
+            .overlay(alignment: .center) {
+                // Sets indication when geoprocessingInProgress is true.
+                if geoprocessingInProgress == true {
+                    ProgressView("Geoprocessing in progress…")
+                        .padding()
+                        .background(.ultraThinMaterial)
+                        .cornerRadius(10)
+                        .shadow(radius: 50)
+                }
+            }
+            .task(id: tapScreenPoint) {
+                guard let tapScreenPoint else { return }
+                geoprocessingInProgress = true
+                // Sets current geoprocessingInProgress to false after process geoprocessing
+                // completes.
+                defer { geoprocessingInProgress = false }
+                addGraphic(at: tapScreenPoint)
+                await calculateViewshed(at: tapScreenPoint)
```

Comment:
```suggestion
                // Sets current geoprocessingInProgress to false after process geoprocessing
                // completes.
                defer { geoprocessingInProgress = false }
                addGraphic(at: tapScreenPoint)
                geoprocessingInProgress = true
                await calculateViewshed(at: tapScreenPoint)
```
Not a big deal, but it is better to set the Boolean right prior to where the async method is called.